mobile/src/main/java/rs/readahead/washington/mobile/views/settings/SecuritySettings.kt

```diff
@@ -248,17 +261,63 @@ class SecuritySettings : BaseFragment() {
     }
 
     private fun setUpLockTimeoutText() {
-        binding.lockTimeoutSetting.setText(lockTimeoutManager.selectedStringRes)
+        binding.lockSettingsButton.setLabelText(getString(lockTimeoutManager.selectedStringRes))
     }
 
     private fun onFailedAttemptChoosen(option: Long) {
         failedUnlockManager.setFailedUnlockOption(option)
         TellaKeysUI.setNumFailedAttempts(option)
-        setUpDeleteAfterUnlockText()
+        TellaKeysUI.setRemainingAttempts(failedUnlockManager.getUnlockRemainingAttempts())
+        setUpSettingsVisibility()
     }
 
-    private fun setUpDeleteAfterUnlockText() {
-        binding.deleteUnlockSetting.setText(failedUnlockManager.getFailedUnlockOption())
+    private fun setUpSettingsVisibility() {
+        if (!CamouflageManager.getInstance().isDefaultLauncherActivityAlias) {
+            with(binding) {
+                binding.unlockRemainingSwitch.isVisible = false
+                camouflageSettingsButton.setOnClickListener {
+                    goToUnlockingActivity(
+                        ReturnActivity.CAMOUFLAGE
+                    )
+                }
+                if (cm.getLauncherName(baseActivity) != null) {
+                    camouflageSettingsButton.setLabelText(cm.getLauncherName(baseActivity))
+                }
+                deleteUnlockSettingsButton.apply {
+                    isBottomLineVisible(failedUnlockManager.getOption() != 0L)
+                    setInfoText(getString(R.string.Settings_feature_not_available_camouflage))
+                    setLabelText(getString(R.string.Settings_Disabled))
+                    setLabelColor(R.color.wa_white_64)
+                    setOnClickListener(null)
+                }
+
+            }
+
```

Comment:
let is try to simplify this code with breaking down the logic into smaller functions

Reply:
what is your opinion @wafalakhal ?